tests/SpecTests/ClientSideEncryptionSpecTest.php

```diff
@@ -1029,6 +1029,43 @@ static function (self $test, ClientEncryption $clientEncryption, ClientEncryptio
         ];
     }
 
+    /**
+     * Prose test 8: Bypass Spawning mongocryptd (via loading shared library)
+     *
+     * @see https://github.com/mongodb/specifications/blob/master/source/client-side-encryption/tests/#via-loading-shared-library
+     */
+    public function testBypassSpawningMongocryptdViaLoadingSharedLibrary(): void
+    {
+        if (! static::isCryptSharedLibAvailable()) {
+            $this->markTestSkipped('Bypass spawning of mongocryptd cannot be tested when crypt_shared is not available');
+        }
+
+        $autoEncryptionOpts = [
+            'keyVaultNamespace' => 'keyvault.datakeys',
+            'kmsProviders' => [
+                'local' => ['key' => new Binary(base64_decode(self::LOCAL_MASTERKEY), 0)],
+            ],
+            'schemaMap' => [
+                'db.coll' => $this->decodeJson(file_get_contents(__DIR__ . '/client-side-encryption/external/external-schema.json')),
+            ],
+            'extraOptions' => [
+                'mongocryptdBypassSpawn' => true,
+                'mongocryptdURI' => 'mongodb://localhost:27021/db?serverSelectionTimeoutMS=1000',
+                'mongocryptdSpawnArgs' => ['--pidfilepath=bypass-spawning-mongocryptd.pid', '--port=27021'],
+                'cryptSharedLibRequired' => true,
+            ],
+        ];
+
+        $clientEncrypted = static::createTestClient(null, [], ['autoEncryption' => $autoEncryptionOpts]);
+
+        $clientEncrypted->selectCollection('db', 'coll')->insertOne(['unencrypted' => 'test']);
+
+        $clientMongocryptd = static::createTestClient('mongodb://localhost:27021');
+
+        $this->expectException(ConnectionTimeoutException::class);
+        $clientMongocryptd->selectDatabase('db')->command(['ping' => 1]);
```

Comment:
Calling `$clientMongocryptd->getManager()->selectServer()` would be the most direct way to trigger a handshake. If you stick with a command, I think `hello` might be more correct (prose tests says "run a handshake command"). I have no idea if mongocryptd supports `ping`, although that's arguably a moot point as the test will never get that far.

```suggestion
        $clientMongocryptd->selectDatabase('db')->command(['hello' => 1]);
```

Reply:
Using `selectServer` sounds good, as that will definitely trigger a handshake without relying on a specific command.